SweetTweetApp/src/main/java/servletsForUsers/ServletUsers.java

```diff
@@ -0,0 +1,58 @@
+package servletsForUsers;
+
+import org.apache.commons.io.IOUtils;
+import users.User;
+import users.UserCollection;
+import users.UserPatch;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.util.Optional;
+
+public class ServletUsers extends HttpServlet {
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        UserPatch userPatch = UserPatch.fromJson(IOUtils.toString(request.getReader()));
+        UserCollection postCollection = new UserCollection();
+        String passwordMaybe = userPatch.getPassword();
+        String passwordReal = UserCollection.hasUser(userPatch.getUsername());
+        response.setContentType("application/json");
+        if (passwordReal != null) {
```

Comment:
In prod, they usually do not distinguish between "wrong password" and "no such user" nowadays.
"Password is wrong" is some piece of information that can be used with malicious goals. That means: 1) such user is actually existing; 2) there's a chance to brute-force p/w or to get in touch with user and phish it out of him/her.

So t is usually "no such user or passsword is invalid" message